flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -96,23 +104,24 @@ private void observeJmDeployment(
             if (status != null
                     && status.getAvailableReplicas() != null
                     && spec.getReplicas().intValue() == status.getReplicas()
-                    && spec.getReplicas().intValue() == status.getAvailableReplicas()
-                    && flinkService.isJobManagerPortReady(effectiveConfig)) {
-
-                // typically it takes a few seconds for the REST server to be ready
+                    && spec.getReplicas().intValue() == status.getAvailableReplicas()) {
+                if (flinkService.isJobManagerPortReady(effectiveConfig)) {
```

Comment:
I don't really see the point in moving this into a nested if branch with fall-through logic. I think the simple if-else before was correct.